pkg/kubenest/controller/global.node.controller/global_node_lifecycle_controller_test.go

```diff
@@ -0,0 +1,348 @@
+							Type:               v1.NodeReady,
+							LastHeartbeatTime:  metav1.NewTime(time.Now().Add(-1 * time.Hour)),
+							LastTransitionTime: metav1.NewTime(time.Now().Add(-1 * time.Hour)),
+						},
+					},
+				},
+			},
+			nodeList: []*v1alpha1.GlobalNode{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "node5",
+					},
+					Status: v1alpha1.GlobalNodeStatus{
+						Conditions: []v1.NodeCondition{
+							{
+								Type:               v1.NodeReady,
+								LastHeartbeatTime:  metav1.NewTime(time.Now().Add(-3 * time.Hour)),
+								LastTransitionTime: metav1.NewTime(time.Now().Add(-3 * time.Hour)),
+							},
+						},
+					},
+				},
+			},
+			expectedStatus: "NotReady",
+		},
+		{
+			name: "Node added to list but with no conditions",
+			initialNode: &v1alpha1.GlobalNode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node6",
+				},
+				Status: v1alpha1.GlobalNodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:               v1.NodeReady,
+							LastHeartbeatTime:  metav1.NewTime(time.Now().Add(-30 * time.Minute)),
+							LastTransitionTime: metav1.NewTime(time.Now().Add(-30 * time.Minute)),
+						},
+					},
+				},
+			},
+			nodeList: []*v1alpha1.GlobalNode{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "node6",
+					},
+					Status: v1alpha1.GlobalNodeStatus{
+						Conditions: []v1.NodeCondition{},
+					},
+				},
+			},
+			expectedStatus: "Unknown",
+		},
+		{
+			name: "Multiple nodes with mixed statuses",
+			initialNode: &v1alpha1.GlobalNode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node7",
+				},
+				Status: v1alpha1.GlobalNodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:               v1.NodeReady,
+							LastHeartbeatTime:  metav1.NewTime(time.Now().Add(-1 * time.Hour)),
+							LastTransitionTime: metav1.NewTime(time.Now().Add(-1 * time.Hour)),
+						},
+					},
+				},
+			},
+			nodeList: []*v1alpha1.GlobalNode{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "node7",
+					},
+					Status: v1alpha1.GlobalNodeStatus{
+						Conditions: []v1.NodeCondition{
+							{
+								Type:               v1.NodeReady,
+								LastHeartbeatTime:  metav1.NewTime(time.Now().Add(-2 * time.Hour)),
+								LastTransitionTime: metav1.NewTime(time.Now().Add(-2 * time.Hour)),
+							},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "node8",
+					},
+					Status: v1alpha1.GlobalNodeStatus{
+						Conditions: []v1.NodeCondition{
+							{
+								Type:               v1.NodeReady,
+								LastHeartbeatTime:  metav1.NewTime(time.Now().Add(-10 * time.Minute)),
+								LastTransitionTime: metav1.NewTime(time.Now().Add(-10 * time.Minute)),
+							},
+						},
+					},
+				},
+			},
+			expectedStatus: "NotReady",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fmt.Println(tt.initialNode.Name + "success")
+			updateStatusForGlobalNode1(tt.nodeList, tt.initialNode)
```

Comment:
The code should not be rewritten for unit testing, and the function name should not be in a form like "updateStatusForGlobalNode1," but should instead be more logical.